packages/docusaurus/src/choosePort.ts

```diff
@@ -116,15 +116,17 @@ export default async function choosePort(
           inquirer.prompt(question).then((answer: any) => {
             if (answer.shouldChangePort) {
               resolve(port);
-            } else {
-              resolve(null);
             }
+            reject(new Error(`Could not run app on port ${defaultPort}.`));
```

Comment:
I can replace the `reject` with `return null`, but this is generally a bad pattern.